File: sdk/php/src/Modules/Patients.php
<?php

namespace OpesCare\Modules;

use OpesCare\Http\ApiClient;

/**
 * Patient record reads — requires valid consent grant.
 */
class Patients
{
    public function __construct(private readonly ApiClient $client) {}

    /**
     * Retrieve consented patient summary: demographics, allergies, medications, labs, visits.
     *
     * Requires consent scope: patients:read
     */
    public function getSummary(string $healthId): array
    {
        return $this->client->get("api/v1/connect/patients/{$healthId}/summary");
    }

    /**
     * Emergency profile access. Requires emergency:access scope and is fully audited.
     */
    public function getEmergencyProfile(string $healthId): array
    {
        return $this->client->get("api/v1/connect/patients/{$healthId}/emergency-profile");
    }

    /**
     * Search patients by Health ID, demographics, national ID, or phone hash.
     */
    public function search(array $params): array
    {
        return $this->client->post('api/v1/connect/patients/search', $params);
    }
}

File: sdk/python/src/opescare/modules/patients.py
from __future__ import annotations

from ..http.api_client import ApiClient


class Patients:
    """Patient record reads — requires valid consent grant."""

    def __init__(self, client: ApiClient) -> None:
        self._client = client

    def get_summary(self, health_id: str) -> dict:
        """
        Consented patient summary: demographics, allergies, medications, labs, visits.
        Requires consent scope: patients:read
        """
        return self._client.get(f"api/v1/connect/patients/{health_id}/summary")

    def get_emergency_profile(self, health_id: str) -> dict:
        """Emergency profile — requires emergency:access scope, fully audited."""
        return self._client.get(f"api/v1/connect/patients/{health_id}/emergency-profile")

    def search(self, **params: object) -> dict:
        """Search patients by health_id, demographics, national_id, or phone."""
        return self._client.post("api/v1/connect/patients/search", dict(params))
